modules/datastore/src/Commands/DictionaryCommands.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\datastore\Commands;
+
+use Drupal\datastore\Service\ResourceProcessor\DictionaryEnforcer;
+use Drupal\datastore\Service\ResourceProcessor\ResourceDoesNotHaveDictionary;
+use Drupal\metastore\DataDictionary\DataDictionaryDiscoveryInterface;
+use Drupal\metastore\ResourceMapper;
+use Drush\Commands\DrushCommands;
+use Drush\Psysh\DrushCommand;
+
+/**
+ * Dictionary Drush commands.
+ *
+ * @codeCoverageIgnore
+ */
+class DictionaryCommands extends DrushCommands {
+
+  /**
+   * Resource mapper service.
+   *
+   * @var \Drupal\metastore\ResourceMapper
+   */
+  protected ResourceMapper $resourceMapper;
+
+  /**
+   * Dictionary enforcer service.
+   *
+   * @var \Drupal\datastore\Service\ResourceProcessor\DictionaryEnforcer
+   */
+  protected DictionaryEnforcer $dictionaryEnforcer;
+
+  /**
+   * Data dictionary discovery service.
+   *
+   * @var \Drupal\metastore\DataDictionary\DataDictionaryDiscoveryInterface
+   */
+  protected DataDictionaryDiscoveryInterface $dataDictionaryDiscovery;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    ResourceMapper $resourceMapper,
+    DictionaryEnforcer $dictionaryEnforcer,
+    DataDictionaryDiscoveryInterface $dataDictionaryDiscovery,
+  ) {
+    parent::__construct();
+    $this->resourceMapper = $resourceMapper;
+    $this->dictionaryEnforcer = $dictionaryEnforcer;
+    $this->dataDictionaryDiscovery = $dataDictionaryDiscovery;
+  }
+
+  /**
+   * Apply the configured data dictionary to a datastore resource.
+   *
+   * @param string $resource_identifier
+   *   Datastore resource identifier, e.g., "b210fb966b5f68be0421b928631e5d51".
+   *
+   * @command dkan:datastore:apply-dictionary
+   */
+  public function applyDictionary(string $resource_identifier) {
+    if ($this->dataDictionaryDiscovery->getDataDictionaryMode() === DataDictionaryDiscoveryInterface::MODE_NONE) {
+      $this->logger()->notice('This site is not configured to use data dictionaries.');
+      return DrushCommand::SUCCESS;
+    }
+    try {
+      $this->dictionaryEnforcer->process(
+        $this->resourceMapper->get($resource_identifier)
```

Comment:
I believe we need to provide the $perspective as an argument because the table name generation is based on the resource_identifier, perspective and version. Perspective if not provided would default to source. In my testing the applyDictionary failed with the following:

```
  SQLSTATE[42S02]: Base table or view not found: 1146 Table 'db.datastore_9683ce1b537c79c9ef21c7f998819433' doe  
  sn't exist: SHOW FULL COLUMNS FROM "datastore_9683ce1b537c79c9ef21c7f998819433"; Array
```
  
  Once I passed in the perspective ie: ResourceLocalizer::LOCAL_FILE_PERSPECTIVE the data dictionary applied correctly.

Reply:
example:

```
    try {
      $this->dictionaryEnforcer->process(
        $this->resourceMapper->get($resource_identifier, ResourceLocalizer::LOCAL_FILE_PERSPECTIVE)
      );
    }
```